File: python/allergies/allergies.py
class Allergies:

    allergens = { 
        "eggs" :1, 
        "peanuts" :2, 
        "shellfish" :4, 
        "strawberries" :8, 
        "tomatoes" :16, 
        "chocolate" :32, 
        "pollen" :64, 
        "cats" :128 
        }

    def __init__(self, score):
        self.score = score

    def allergic_to(self, item):
        return Allergies.allergens[item] & self.score != 0

    @property
    def lst(self):
        res = []

        for key in Allergies.allergens.keys():
            if self.allergic_to(key):
                res.append(key)

        return res

File: nim/allergies/allergies.nim
import std/bitops, std/math, std/sequtils, std/setutils

type
  Allergen* = enum
    Eggs, Peanuts, Shellfish, Strawberries, Tomatoes, Chocolate, Pollen, Cats

proc isAllergicTo*(score: int, allergen: Allergen): bool =
  let code = pow(2.0, float(allergen.ord))
  bitand(score, int(code)) > 0

proc allergies*(score: int): set[Allergen] =
    Allergen.toSeq
            .filter(proc(allergen: Allergen): bool = isAllergicTo(score, allergen))
            .toSet
